File: File Templates/Custom/UICollectionReusableView.xctemplate/UICollectionReusableViewXIBObjective-C/___FILEBASENAME___.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

/* System */

/* ViewController */

/* View */

/* Model */

/* Util */

/* NetWork InterFace */

/* Vender */

@interface ___FILEBASENAMEASIDENTIFIER___ ()

@end

@implementation ___FILEBASENAMEASIDENTIFIER___

#if DEBUG
- (void)dealloc {
    NSLog(@"[%@ dealloc]", NSStringFromClass(self.class));
}
#endif

#pragma mark - life cycle
- (instancetype)initWithFrame:(CGRect)frame {
    if ((self = [super initWithFrame:frame])) {
        [self commonInit];
    }
    return self;
}
- (void)awakeFromNib {
    [super awakeFromNib];
    [self commonInit];
}

#pragma mark - initial Methods
- (void)commonInit {
    /*custom view u want draw in here*/
    self.backgroundColor = [UIColor whiteColor];
    [self addSubViews];
    [self addSubViewConstraints];
}

#pragma mark - add subview
- (void)addSubViews {

}

#pragma mark - layout
- (void)addSubViewConstraints {

}

#pragma mark - private method

#pragma mark - public method

#pragma mark - getters and setters

@end

File: File Templates/Custom/UICollectionReusableView.xctemplate/UICollectionReusableViewXIBSwift/___FILEBASENAME___.swift
//___FILEHEADER___

/* System */
import UIKit

/* ViewController */

/* View */

/* Model */

/* Util */

/* NetWork InterFace */

/* Vender */

class ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_cocoaTouchSubclass___ {

    //MARK: - public property

    //MARK: - readonly property

    //MARK: - getters and setters


    //MARK: - life cycle
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.commonInit()
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }

    override func awakeFromNib() {
        super.awakeFromNib()
        self.commonInit()
    }

    #if DEBUG
    deinit {
        print("[\(type(of: self)) deinit]")
    }
    #endif
}


//MARK: - initial Methods
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {
    func commonInit() {
        self.backgroundColor = UIColor.white;
        self.addSubViews()
        self.addSubViewConstraints()

    }
}

//MARK: - add Subview
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {

    func addSubViews() {

    }
}

//MARK: - layout
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {

    func addSubViewConstraints() {

    }
}

//MARK: - private method
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {

}

//MARK: - public method
extension ___FILEBASENAMEASIDENTIFIER___ {

}
